src/serde.rs

```diff
@@ -0,0 +1,153 @@
+use crate::qdrant::value::Kind;
+use crate::qdrant::{ListValue, Struct, Value};
+use serde::ser::{SerializeMap, SerializeSeq};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::collections::HashMap;
+
+impl Serialize for Value {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if let Some(value) = &self.kind {
+            match value {
+                Kind::NullValue(_v) => serializer.serialize_none(),
+                Kind::DoubleValue(double) => serializer.serialize_f64(*double),
+                Kind::IntegerValue(int) => serializer.serialize_i64(*int),
+                Kind::StringValue(str) => serializer.serialize_str(str),
+                Kind::BoolValue(bool) => serializer.serialize_bool(*bool),
+                Kind::StructValue(struc) => {
+                    let mut map = serializer.serialize_map(Some(struc.fields.len()))?;
+                    for (k, v) in struc.fields.iter() {
+                        map.serialize_entry(k, v)?;
+                    }
+                    map.end()
+                }
+                Kind::ListValue(list) => {
+                    let mut seq = serializer.serialize_seq(Some(list.values.len()))?;
+                    for v in &list.values {
+                        seq.serialize_element(v)?;
+                    }
+                    seq.end()
+                }
+            }
+        } else {
+            serializer.serialize_none()
+        }
+    }
+}
+
+impl From<serde_json::Value> for Value {
+    fn from(value: serde_json::Value) -> Self {
+        match value {
+            serde_json::Value::Null => Self {
+                kind: Some(Kind::NullValue(0)),
+            },
+            serde_json::Value::Bool(bool) => Self {
+                kind: Some(Kind::BoolValue(bool)),
+            },
+            serde_json::Value::Number(num) => {
+                if let Some(num) = num.as_i64() {
+                    Self {
+                        kind: Some(Kind::IntegerValue(num)),
+                    }
+                } else if let Some(num) = num.as_f64() {
+                    Self {
+                        kind: Some(Kind::DoubleValue(num)),
+                    }
+                } else {
+                    Self {
+                        kind: Some(Kind::DoubleValue(0.0)),
+                    }
```

Comment:
I guess this shouldn't really happen in practice, but is zero really the right default here? If we don't want to panic, at least default to NaN, so that people can figure out something's wrong.

Reply:
:+1: